python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -432,8 +432,9 @@ def _get_run_tags(self, run_id: str) -> Tags:
     def _get_command_args(self, run_args: ExecuteRunArgs, context: LaunchRunContext):
         return run_args.get_command_args()
 
-    @staticmethod
-    def get_image_for_run(run: DagsterRun) -> Optional[str]:
+    def get_image_for_run(self, context: LaunchRunContext) -> Optional[str]:
+        """Child classes can override this method to determine the image to use for a run. This is considered a public API."""
```

Comment:
I would remove the "considered a public API" part - I think we still reserve the right to change it in the future if needed